storage/blob.go

```diff
@@ -595,13 +595,51 @@ func (b BlobStorageClient) leaseCommonPut(container string, name string, headers
 	return resp.headers, nil
 }
 
+// SnapshotBlob creates a snapshot for a blob as per https://msdn.microsoft.com/en-us/library/azure/ee691971.aspx
+func (b BlobStorageClient) SnapshotBlob(container string, name string, timeout int, currentLeaseID string) (snapshotTimestamp *time.Time, err error) {
```

Comment:
Add the `extraHeaders` parameter present in other functions (actually, leaseID could go there too). Check how this works with other funcs, for example `PutAppendBlob`